src/main/org/nlogo/prim/etc/_extracthsb.scala

```diff
@@ -0,0 +1,99 @@
+// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo
+
+package org.nlogo.prim.etc
+
+import org.nlogo.api, api.{ Syntax, LogoListBuilder, LogoList }
+import org.nlogo.nvm.{ Context, Pure, Reporter }
+
+object extracthsb {
```

Comment:
The goal is to keep language-independent engine code from code that is specific to the NetLogo language. If the code would be useful for writing agent-based models in another JVM language besides NetLogo, then it shouldn't be in the `prim` package. The code in a `prim` class should marshal arguments, check argument types, and report errors, but it shouldn't perform the actual job of the primitive (unless it's something that every language has, so e.g. it would be silly to require that `_plus` call separate code to add the numbers; we might as well just inline it). The actual meat should be in `agent` or a package `agent` depends on such as `api`.